src/tools/wasm2js.cpp

```diff
@@ -353,11 +353,12 @@ static void optimizeJS(Ref ast) {
     } else if (isUnary(node, L_NOT)) {
       node[2] = optimizeBoolean(node[2]);
     }
-    // Add/subtract can merge coercions up.
+    // Add/subtract can merge coercions up, except for divisions
```

Comment:
please mention in the comment that divisions are an exception because of the fraction issue